app/data_model/answer_store.py

```diff
@@ -0,0 +1,133 @@
+import re
+
+from collections import OrderedDict
+
+
+class AnswerStore(object):
+    """
+    An object that stores the flattened structure of AnswerStates.
+    It is referenced by the answers property in the QuestionnaireStore.
+    """
+
+    def __init__(self, existing_answers=None):
+        self.answers = existing_answers or []
+
+    def add(self, answer):
+        """
+        Add a new answer into the answer store.
+
+        :param answer: A dict of flattened answer details.
+        :return: None.
+        """
+        answer_to_add = answer.copy()
+
+        if self.exists(answer_to_add):
```

Comment:
Does it make sense to merge the add/update functionality. It makes update kind of redundant.